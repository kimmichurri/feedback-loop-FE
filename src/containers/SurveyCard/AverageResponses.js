var avg = {
  "averages": [
    {
      "question_id": 42,
      "questionTitle": "Pick a number between one and four",
      "average_rating": 3.3333333333333333
    }
  ],
  "survey": {
    "id": 1,
    "surveyName": "Test survey",
    "surveyExpiration": null,
    "created_at": "2019-05-23T05:24:58",
    "updated_at": "2019-05-23T05:24:58",
    "status": "active",
    "questions": [
      {
        "id": 42,
        "questionTitle": "Pick a number between one and four",
        "options": [
          {"description": "Four", "value": 4},
          {"description": "Three", "value": 3},
          {"description": "Two", "value": 2},
          {"description": "One", "value": 1}
        ]
      }
    ],
    "groups": [
      {
        "name": "Team1",
        "members": [{
          "id": 7,
          "name": "Peter Lapicola",
          "cohort": "1811",
          "program": "B",
          "status": "Active"
        },
        {
          "id": 8,
          "name": "April Dagonese",
          "cohort": "1811",
          "program": "B",
          "status": "Active"
        },
        {
          "id": 9,
          "name": "Scott Thomas",
          "cohort": "1811",
          "program": "B",
          "status": "Active"
        }
      ],
    }
  ]
  }
}

export default avg;
