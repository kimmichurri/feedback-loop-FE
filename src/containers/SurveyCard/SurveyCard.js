import React, { Component } from 'react';
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

export class SurveyCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dataDisplay: "none"
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
  }

  handleChange = (e) => {
  }

  toggleData = () => {
    let display
    this.state.dataDisplay == "none"
    ? display = "flex"
    : display = "none"

    this.setState({
      dataDisplay: display
    })
  }

  prettyDate = (dateString) => {
    let date = new Date(dateString)
    date = date.toLocaleDateString()
    return date
  }

  render() {
    return(
      <section className='survey-accordion'>
        <div className='survey-card'
             onClick={this.toggleData}>
          <article className='survey-card-name'>
            {this.props.surveyData.surveyName}
          </article>
          <article className='card-exp-date'>
            {this.prettyDate(this.props.surveyData.surveyExpiration)}
          </article>
        </div>
        <div className='survey-data'
             style={{display: this.state.dataDisplay}}>
          BUTTS
        </div>
      </section>
    )
  }
}

export default SurveyCard
