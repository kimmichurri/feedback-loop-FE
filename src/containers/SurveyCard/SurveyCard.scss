.survey-card {
  border-top: .5rem solid #12f1fc;
  -webkit-box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
  box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
  color: #5f5f5f;
  cursor: pointer;
  padding: 1rem;
  background-color: white;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.survey-data {
  color: white;
  background-color: $light-grey;
  width: 100%;
  padding: 1rem;
  box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
  transition: ease .4s all;
  animation-name: fadeIn;
  animation-duration: .4s;
  animation-fill-mode: both;
  will-change: transform;
}

.survey-accordion {
  margin-bottom: 1%;
  width: 80%;
  box-shadow: 0 0 2px 1px rgba(0, 0, 0, 0.2);
}
