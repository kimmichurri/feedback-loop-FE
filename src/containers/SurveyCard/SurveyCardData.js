import React, { Component } from 'react';
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import UserAvg from './UserAverages'
import AvgResp from './AverageResponses'

export class SurveyCardData extends Component {
  constructor(props) {
    super(props);
    this.state = {
      averages: '',
      userAverages: ''
    }
  }

  componentDidMount = async () => {
    // avgsUrl = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.sData.id}/averages`
    // avgsResponse = await fetch(avgsUrl)
    // averages = await avgsResponse.json()
    //
    // userAvgsUrl = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.sData.id}/user_averages`
    // userAvgsResponse = await fetch(userAvgsUrl)
    // userAverages = await userAvgsResponse.json()

    this.setState({
      averages: AvgResp,
      userAverages: UserAvg
    })
  }


  displayGroup = (group) => {
    return(
      <section className="s-group">{group.name}
      </section>
    )
  }

  render() {
    return(
      <>
        <section className="s-status">{this.props.sData.status}</section>
        <section className="s-groups">{this.props.sData.groups.map(group => {
          return this.displayGroup(group)
        })}
        </section>
      </>
    )
  }
}

export default SurveyCardData
