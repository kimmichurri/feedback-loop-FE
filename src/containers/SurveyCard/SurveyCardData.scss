.s-status {
  align-self: flex-start;
  padding-bottom: 1%;
  color: $accent-charcoal;
}

.s-groups {
  margin-left: 10%;
  margin-right: -15px;
}

.s-group {
  padding-top: 2%;
  padding-bottom: 2%;
  border-top-left-radius: 12px;
  border-bottom-left-radius: 12px;
  background-color: white;
  color: black;
  box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
}

.s-group-name {
  text-align: left;
  padding-left: 2%;
}

.s-question, .user-rating, .q-avg-rating {
  text-align: end;
  padding-right: 2%;
}

.s-question-data, .s-question {
  margin-bottom: 1%;
}

.s-question {
  color: $accent-red;
  font-weight: bold;
}
