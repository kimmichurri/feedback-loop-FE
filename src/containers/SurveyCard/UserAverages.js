var avg = {
  "averages": [
    {
      "average_rating": 3.5000000000000000,
      "question_id": 42,
      "user_id": 1
    },
    {
      "average_rating": 3.0000000000000000,
      "question_id": 42,
      "user_id": 2
      }
    ],
    "survey": {
      "created_at": "2019-05-23T05:24:58",
      "surveyExpiration": null,
      "groups": [
        {
          "name": "Test",
          "members": [
            {
              "id": 1,
              "name": "Peter Lapicola",
              "cohort": "1811",
              "program": "B",
              "status": "Active"
            },
            {
              "id": 2,
              "name": "April Dagonese",
              "cohort": "1811",
              "program": "B",
              "status": "Active"
            },
            {
              "id": 3,
              "name": "Scott Thomas",
              "cohort": "1811",
              "program": "B",
              "status": "Active"
            }
          ]
        }
      ],
      "id": 1,
      "surveyName": "Test Survey",
      "questions": [
        {
          "options": [
            {
              "description": "Four",
              "pointValue": 4
            },
            {
              "description": "Three",
              "pointValue": 3
            },
            {
              "description": "Two",
              "pointValue": 2
            },
            {
              "description": "One",
              "pointValue": 1
            }
          ],
        "id": 42,
        "questionTitle": "Pick a number between one and four"
      }
    ],
    "status": "active",
    "updated_at": "2019-05-23T05:24:58"
  }
}

export default avg;
