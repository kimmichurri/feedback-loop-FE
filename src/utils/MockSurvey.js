var survey = {
    "created_at": "2019-05-26T17:18:25",
    "groups": [
        {
            "members": [
                {
                    "cohort": "1811",
                    "id": 3,
                    "name": "Scott Thomas",
                    "program": "B",
                    "status": "Active"
                },
                {
                    "cohort": "1811",
                    "id": 8,
                    "name": "Jessica Hansen",
                    "program": "F",
                    "status": "Active"
                },
                {
                    "cohort": "1811",
                    "id": 2,
                    "name": "April Dagonese",
                    "program": "B",
                    "status": "Active"
                }
            ],
            "name": "1811"
        }
    ],
    "id": 21,
    "questions": [
        {
            "id": 19,
            "options": [
                {
                    "description": "4",
                    "id": 76,
                    "pointValue": 4
                },
                {
                    "description": "3",
                    "id": 75,
                    "pointValue": 3
                },
                {
                    "description": "2",
                    "id": 74,
                    "pointValue": 2
                },
                {
                    "description": "1",
                    "id": 73,
                    "pointValue": 1
                }
            ],
            "questionTitle": "How good is this jabroni?"
        }
    ],
    "status": "Closed",
    "surveyExpiration": "2019-05-26T17:17:42",
    "surveyName": "Inst Test",
    "updated_at": "2019-05-27T21:40:28"
}

export default survey
