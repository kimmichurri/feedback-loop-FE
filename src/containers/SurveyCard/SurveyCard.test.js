import React from 'react'
import { shallow } from 'enzyme'
import { SurveyCard, mapStateToProps, mapDispatchToProps } from './SurveyCard'
import MockSurvey from './MockSurvey'
jest.mock('../../thunks/handlePost')

describe('SurveyCard', () => {
  let mockState
  let wrapper
  let mockEvent

  beforeEach(() => {
    mockState = {
      dataDisplay: "none"
    }
    wrapper = shallow(
      <SurveyCard
        surveyData={MockSurvey}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have proper default state', () => {
    expect(wrapper.state()).toEqual({
      dataDisplay: "none"
    })
  })

  it('should set toggle the data display', () => {
    wrapper.instance().toggleData()

    expect(wrapper.state()).toEqual({
      dataDisplay: "flex"
    })
  })
})
