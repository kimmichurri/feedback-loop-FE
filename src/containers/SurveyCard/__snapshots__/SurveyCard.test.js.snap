// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyCard should match the snapshot 1`] = `
<section
  className="survey-accordion"
>
  <div
    className="survey-card"
    onClick={[Function]}
    style={
      Object {
        "border-top": ".5rem solid #13F1FC",
      }
    }
  >
    <article
      className="survey-card-name"
    >
      Inst Test
    </article>
    <article
      className="card-exp-date"
    >
      5/26/2019
    </article>
  </div>
  <div
    className="survey-data"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SurveyCardData
      key="21"
      sData={
        Object {
          "created_at": "2019-05-26T17:18:25",
          "groups": Array [
            Object {
              "members": Array [
                Object {
                  "cohort": "1811",
                  "id": 3,
                  "name": "Scott Thomas",
                  "program": "B",
                  "status": "Active",
                },
                Object {
                  "cohort": "1811",
                  "id": 8,
                  "name": "Jessica Hansen",
                  "program": "F",
                  "status": "Active",
                },
                Object {
                  "cohort": "1811",
                  "id": 2,
                  "name": "April Dagonese",
                  "program": "B",
                  "status": "Active",
                },
              ],
              "name": "1811",
            },
          ],
          "id": 21,
          "questions": Array [
            Object {
              "id": 19,
              "options": Array [
                Object {
                  "description": "4",
                  "id": 76,
                  "pointValue": 4,
                },
                Object {
                  "description": "3",
                  "id": 75,
                  "pointValue": 3,
                },
                Object {
                  "description": "2",
                  "id": 74,
                  "pointValue": 2,
                },
                Object {
                  "description": "1",
                  "id": 73,
                  "pointValue": 1,
                },
              ],
              "questionTitle": "How good is this jabroni?",
            },
          ],
          "status": "Closed",
          "surveyExpiration": "2019-05-26T17:17:42",
          "surveyName": "Inst Test",
          "updated_at": "2019-05-27T21:40:28",
        }
      }
    />
  </div>
</section>
`;
